sdk/search/search-documents/test/utils/setup.ts

```diff
@@ -443,6 +443,11 @@ export async function deleteDataSourceConnections(client: SearchIndexerClient):
 
 // eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
 export async function createSkillsets(client: SearchIndexerClient): Promise<void> {
+  const skillSets = await client.listSkillsets();
```

Comment:
so the fix is basically to cleanup previously created assets?

I worry a little bit that someone might run these tests against a Search instance with existing content. Can we perhaps check that the name begins with `my-azureblob-skillset` or some other way to know it's one we created before we delete it? We could throw an exception if we see an entity other than one we created in a previous run.

Reply:
Yes it might delete existing resources. But, throwing errors will not help in the case of CI r8? Also, this does not delete indexes. Just indexers, datasources & Synonymmaps whose use case is very minimal and going to impact only our team members. (Our subscription allows only a max of 3-5 in each case. So, recreating them, if any, should be relatively easy)

Reply:
This isn't just for CI though, according to our own documentation:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/search/search-documents/test/README.md

> You can use existing Azure resources for the live tests, or generate new ones by using our New-TestResources.ps1 script

If we're encouraging developers to run tests, and we're telling them how to run the tests, and we're saying it's OK to point the tests at an existing Search subscription.... how is it OK to delete things we didn't create?

@tg-msft @heaths how does .NET handle this kind of cleanup?

Reply:
I think we should assume contributors self-solving a problem could be running tests against resources with data they care about. Its a very bad idea (and I think the Readme should call this out!), but reasonable precautions against destroying data we didn't create seems like a good idea.

Reply:
In .NET, we have a class that we use to create - or otherwise keep track of - any resources we create. Could you do something like that here? We use `IAsyncDisposable` so that it always runs even in the case of an exception. Could you use `after` or `afterEach` with mocha to accomplish the same?